doc/how-to/app/app_roles.rst

```diff
@@ -0,0 +1,476 @@
+
+        *   -   **Members**
+            -
+
+        *   -   :ref:`validate([cfg]) <roles_api_reference_validate>`
+            -   Validate a role's configuration.
+
+        *   -   :ref:`apply([cfg]) <roles_api_reference_apply>`
+            -   Apply a role's configuration.
+
+        *   -   :ref:`stop() <roles_api_reference_stop>`
+            -   Stop a role.
+
+        *   -   :ref:`dependencies <roles_api_reference_dependencies>`
+            -   Define a role's dependencies.
+
+
+
+
+.. _roles_api_reference_validate:
+
+..  function:: validate([cfg])
+
+    Validate a role's configuration.
+    This function is called on instance startup or when the :ref:`configuration is reloaded <etcd_reloading_configuration>` for the instance with this role.
+    Note that the ``validate()`` function is called regardless of whether the role's configuration or any field in a cluster's configuration is changed.
+
+    ``validate()`` should throw an error if the validation fails.
+
+    :param cfg: a role's role configuration to be validated.
+                This parameter provides access to configuration options defined in :ref:`roles_cfg.\<role_name\> <configuration_reference_roles_cfg>`.
+                To get values of configuration options placed outside ``roles_cfg.<role_name>``, use :ref:`config:get() <config_api_reference_get>`.
+
+    See also: :ref:`roles_create_custom_role_validate`
+
+
+.. _roles_api_reference_apply:
+
+..  function:: apply([cfg])
+
+    Apply a role's configuration.
+    ``apply()`` is called after ``validate()`` is executed for all the enabled roles.
+    As the ``validate()`` function, ``apply()`` is called on instance startup or when the configuration is reloaded for the instance with this role.
+
+    ``apply()`` should throw an error if the specified configuration can't be applied.
+
+    ..  NOTE::
+
+        Note that ``apply()`` is not invoked if an instance switches to read-write mode when :ref:`replication.failover <configuration_reference_replication_failover>` is set to ``election`` or ``supervised``.
+        You can check an instance state by subscribing to the ``box.status`` event using :ref:`box.watch() <box-watch>`.
+
+    :param cfg: a role's role configuration to be applied.
+                This parameter provides access to configuration options defined in :ref:`roles_cfg.\<role_name\> <configuration_reference_roles_cfg>`.
+                To get values of configuration options placed outside ``roles_cfg.<role_name>``, use :ref:`config:get() <config_api_reference_get>`.
+
+    See also: :ref:`roles_create_custom_role_apply`
+
+
+.. _roles_api_reference_stop:
+
+..  function:: stop()
+
+    Stop a role.
+    This function is called on configuration reload if the role is removed from ``roles`` for the given instance.
+
+    See also: :ref:`roles_create_custom_role_stop`
+
+.. _roles_api_reference_dependencies:
+
+..  data:: dependencies
+
+    (Optional) Define a role's dependencies.
```

Comment:
```suggestion
    (Optional) Role's dependencies.
```
Nit: IMO, verbs are not needed in option/parameter/variable descriptions in references.